app/services/puppet_class_importer.rb

```diff
@@ -254,10 +254,10 @@ def load_classes_from_json(blob)
 
   def add_classes_to_foreman(env_name, klasses)
     env         = find_or_create_env env_name
-    new_classes = klasses.map { |k| Puppetclass.where(:name => k[0]).first_or_create }
+    new_classes = klasses.map { |k| Puppetclass.unscoped.where(:name => k[0]).first_or_create }
```

Comment:
Why unscoped? Doesn't this run on the user's context so it should be scoped by `User.current`?

Reply:
Running this in a scope is exactly the problem. When a puppet class with given name already exists in different scope, it fails to find it but hits the uniqueness constraint when saving a new record.

Reply:
Oh that's a good catch! Can you put it in a comment here so that we're reminded in case we modify this code again?